utils.py

```diff
@@ -275,25 +274,120 @@ def set(self, lru_cache):
 ip_cache = LRUCacheSingleton()
 
 
-def is_ip_per_task(app):
-    """
-    Return whether the application is using IP-per-task.
-    :param app:  The application to check.
-    :return:  True if using IP per task, False otherwise.
-    """
-    return app.get('ipAddress') is not None
+def get_app_networking_mode(app):
+    mode = 'host'
 
+    if app.get('ipAddress'):
+        mode = 'container'
 
-def is_user_network(app):
-    """
-    Returns True if container network mode is set to USER
-    :param app:  The application to check.
-    :return:  True if using USER network, False otherwise.
-    """
-    c = app.get('container', {})
-    return c is not None and c.get('type', '') == 'DOCKER' and \
-        c.get('docker', {})\
-        .get('network', '') == 'USER'
+    _mode = app.get('container', {})\
```

Comment:
TIL:
```
Python 2.7.8 (default, Apr 15 2015, 09:26:43) 
[GCC 4.9.2 20150212 (Red Hat 4.9.2-6)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> foo={'abc': None}
>>> foo.get('abc', {}).get('baz', '')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'get'
```

so, when Marathon returns `null` for a value like `container` the logic here breaks down